src/libraries/System.Private.CoreLib/src/System/Int128.cs

```diff
@@ -2192,5 +2160,27 @@ static bool INumberBase<Int128>.TryConvertToTruncating<TOther>(Int128 value, [Ma
 
         /// <inheritdoc cref="IUnaryPlusOperators{TSelf, TResult}.op_UnaryPlus(TSelf)" />
         public static Int128 operator +(Int128 value) => value;
+
+        //
+        // IBinaryIntegerParseAndFormatInfo
+        //
+
+        static bool IBinaryIntegerParseAndFormatInfo<Int128>.IsSigned => true;
+
+        static bool IBinaryIntegerParseAndFormatInfo<Int128>.IsUnsigned => false;
+
+        static int IBinaryIntegerParseAndFormatInfo<Int128>.MaxDigitCount => 39; // 170_141_183_460_469_231_731_687_303_715_884_105_727
+
+        static int IBinaryIntegerParseAndFormatInfo<Int128>.MaxHexDigitCount => 32; // 0x7FFF_FFFF_FFFF_FFFF_FFFF_FFFF_FFFF_FFFF
+
+        static Int128 IBinaryIntegerParseAndFormatInfo<Int128>.MaxValueDiv10 => new Int128(0x0CCC_CCCC_CCCC_CCCC, 0xCCCC_CCCC_CCCC_CCCC);
+
+        static string IBinaryIntegerParseAndFormatInfo<Int128>.OverflowMessage => SR.Overflow_Int128;
+
+        static bool IBinaryIntegerParseAndFormatInfo<Int128>.IsGreaterThanAsUnsigned(Int128 left, Int128 right) => (UInt128)(left) > (UInt128)(right);
+
+        static Int128 IBinaryIntegerParseAndFormatInfo<Int128>.MultiplyBy10(Int128 value) => value * 10;
+
+        static Int128 IBinaryIntegerParseAndFormatInfo<Int128>.MultiplyBy16(Int128 value) => value * 16;
```

Comment:
No.

We have `IsPositive` and `IsNegative`, but we need to know if the type implements `ISignedNumber<TSelf>` or `IUnsignedNumber<TSelf>`, we can't query the latter due to the recursive constraint and limitations in the language today.

We could compute `MaxDigitCount`, `MaxHexDigitCount`, and `MaxValueDiv10` but that puts more pressure on the JIT and inliner to do the right thing. It is also not likely to happen for `Int128`/`UInt128` as they are user-defined structs, not primitives.

`IsGreaterThanUnsigned` can't be done since we don't have associated types, so we don't have a way to represent something like `(left < 0) || (left  > right)` efficiently. The JIT doesn't do the right thing today, so this allows us to provide it "per type".

`MultiplyBy10` and `MultiplyBy16` allow us to remove an additional `TInteger.CreateTruncating(...)` in the hot paths and will allow us to easily optimize some special cases, like `Int128`, `UInt128`, or `BigInteger`, if that's important in the future.

Reply:
Ok, thanks.